lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -101,12 +102,84 @@ class IntProgCmdProvider: public CmdProvider {
     static const ResponseInfo m_responses[SEQ_LENGHT]; /**< Array of response information */
 };
 
-#if 0
+/**
+ * @class FwProgCmdProvider
+ * @brief Implementation of the command provider for firmware programming.
+ *
+ * This class provides commands for firmware programming, specifically tailored
+ * for the Zumo32U4 device.
+ */
 class FwProgCmdProvider : public CmdProvider {
     public:
-        FwProgCmdProvider(FileHandle * fs);
-}
-#endif
+        /**
+         * @brief Constructor for the FwProgCmdProvider class.
+         * @param fs A pointer to the FileHandler object for handling file operations.
+         */
+        FwProgCmdProvider(FileHandler * fs) :
+        m_index(0),
+        m_fileSystem(fs)
+    {}
+
+    /**
+     * @brief Resets the command provider to its initial state.
+     *
+     * This method resets the internal index that tracks the current command
+     * in the command sequence memory.
+     */
+    virtual void reset() { m_index = 0; }

```

Comment:
expect more a reopen/rewind of a file here.

Reply:
It will be taken care of by the next commit.